contracts/staking.sol

```diff
@@ -68,12 +79,20 @@ contract Staking {
             !_addressToIsValidator[msg.sender] &&
             _addressToStakedAmount[msg.sender] >= ValidatorThreshold
         ) {
-            // append to validator set
-            _addressToIsValidator[msg.sender] = true;
-            _addressToValidatorIndex[msg.sender] = _validators.length;
-            _validators.push(msg.sender);
+            if (_validators.length < MaximumNumValidators) {
+                _appendToValidatorSet(msg.sender);
+            } else {
+                require(
+                    _addressToStakedAmount[msg.sender] > _minimumStakedAmountByValidator,
+                    "Total amount the account staked is less than current minimum staked amount by a validator"
+                );
+
+                _deleteFromValidators(_lowestValidator);
+                _appendToValidatorSet(msg.sender);
+            }
```

Comment:
Actually _appendToValidatorSet is called in both case of 2nd level if-else block. You can remove else block like this:

```sol
           if (_validators.length >= MaximumNumValidators) {
               
                require(
                    _addressToStakedAmount[msg.sender] > _minimumStakedAmountByValidator,
                    "Total amount the account staked is less than current minimum staked amount by a validator"
                );

                _deleteFromValidators(_lowestValidator);
                _appendToValidatorSet(msg.sender);
            }
           _appendToValidatorSet(msg.sender);
```

Reply:
Good catch ! Updated in https://github.com/0xPolygon/staking-contracts/pull/10/commits/fb5f135ad666c6e9b41b33992f1ad10e801dd894